src/general/AutoModLoader.cs

```diff
@@ -0,0 +1,65 @@
+using System.IO;
+using System.Reflection;
+using Godot;
+using Newtonsoft.Json;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+
+public class AutoModLoader : Node
+{
+    private AutoModLoader()
+    {
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
+        foreach (DirectoryInfo currentMod in modFolder.EnumerateDirectories())
```

Comment:
The previous comment likely requires this to be changed to use the Godot engine provided directory iterator.